tap_appstore/client.py

```diff
@@ -18,133 +23,117 @@
+            fieldnames = [col.strip().replace(' ', '_').lower() for col in first_line.split('\t')]
+
+            reader = csv.DictReader(data_io, delimiter='\t', fieldnames=fieldnames)
+
+            for record in reader:
+                first_value = next(iter(record.values()))
+                if first_value and any(keyword in first_value for keyword in self.skip_line_first_values):
+                    logger.info(f"Skipping line: {record}")
+                    continue
+
+                line_id += 1
+                record['_line_id'] = line_id
+                record['_time_extracted'] = datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%SZ')
+                record['_api_report_date'] = start_date
+
+                processed_record = self.post_process(record, context)
+                if processed_record is not None:
+                    yield processed_record
+
+        except APIError as e:
+            logger.error(f'Error during download report {self.name}.\n{e}')
+            raise
+
+    @staticmethod
+    def convert_date(date_str, date_format='%Y-%m-%d'):
+        """Converts date string to ISO format based on the given date format, defaulting to '%Y-%m-%d'.
+           Returns None if the date string is invalid or empty.
         """
-        # TODO: Delete this method if no payload is required. (Most REST APIs.)
-        return None
 
-    def parse_response(self, response: requests.Response) -> Iterable[dict]:
-        """Parse the response and return an iterator of result records.
-
-        Args:
-            response: The HTTP ``requests.Response`` object.
-
-        Yields:
-            Each record from the source.
-        """
-        # TODO: Parse response body and return a set of records.
-        yield from extract_jsonpath(self.records_jsonpath, input=response.json())
-
-    def post_process(
-        self,
-        row: dict,
-        context: dict | None = None,  # noqa: ARG002
-    ) -> dict | None:
-        """As needed, append or transform raw data to match expected structure.
-
-        Args:
-            row: An individual record from the stream.
-            context: The stream context.
-
-        Returns:
-            The updated record dictionary, or ``None`` to skip the record.
-        """
-        # TODO: Delete this method if not needed.
-        return row
+        if not date_str:
+            logger.warning(f"Empty or invalid date string provided; cannot convert using format {date_format}")
+            return None
+
+        date_str = date_str.strip()
+        if not date_str:
+            logger.warning(f"Empty or invalid date string provided; cannot convert using format {date_format}")
+            return None
+
+        try:
+            return datetime.strptime(date_str, date_format).isoformat()
+        except ValueError as e:
+            logger.error(f"Date conversion error for date: '{date_str}' with format '{date_format}' | Error: {str(e)}")
+            return None
```

Comment:
Let's reraise the exception, we can't proceed if we have an error in the datetime.

As an example, if applestore changes the date format we can't ignore these values, we need to fail the extraction so we can fix the schema with the correct format.